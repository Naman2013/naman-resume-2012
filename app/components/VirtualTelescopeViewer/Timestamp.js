import React from 'react';
import PropTypes from 'prop-types';
import InlineCountdown from '../common/inline-countdown';

import { brightGreen } from '../../styles/variables/colors';

const propTypes = {
  unixTimestamp: PropTypes.number,
};

const defaultProps = {
  unixTimestamp: 0,
};

function Timestamp({ unixTimestamp }) {
  if (!unixTimestamp) { return null; }

  return (
    <div className="root">
      <h5 className="content">
        <InlineCountdown
          incrementTime={true}
          startTime={unixTimestamp}
          format={'YYYY-M-DDTH:mmUTC'}
        />
      </h5>

      <style jsx>{`
        .root {
          margin-top: 15px;
          text-align: center;
        }

        :global(.root .inline-countdown) {
          display: inline;
        }

        .content {
          border: 1px solid ${brightGreen};
          padding: 0 5px;
          margin: 0;
          display: inline;
        }
      `}</style>
    </div>
  );
}

Timestamp.propTypes = propTypes;
Timestamp.defaultProps = defaultProps;

export default Timestamp;
