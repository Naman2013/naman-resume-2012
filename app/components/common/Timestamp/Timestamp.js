import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';

const propTypes = {
  timestampInSeconds: PropTypes.number,
};

const defaultProps = {
  timestampInSeconds: 0,
};

const Timestamp = ({ timestampInSeconds }) => {
  if (!timestampInSeconds) return null;
  const convertedTimeStamp = timestampInSeconds * 1000;
  const easternTime = moment.tz(convertedTimeStamp, 'America/New_York').format('h:mma z');
  const pacificTime = moment.tz(convertedTimeStamp, 'America/Los_Angeles').format('h:mma z');
  const utcTime = moment.utc(convertedTimeStamp).format('HH:mm z');

  return (
    <span>{easternTime} · {pacificTime} · {utcTime}</span>
  );
};


Timestamp.propTypes = propTypes;
Timestamp.defaultProps = defaultProps;

export default Timestamp;
