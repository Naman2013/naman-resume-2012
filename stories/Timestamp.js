import React from 'react';
import { storiesOf } from '@storybook/react';
import Timestamp from '../app/components/common/Timestamp';

const START_TIME = 1499570100;

storiesOf('Timestamp', module)
  .add('EST/EDT/PST/PDT UTC', () => (
    <Timestamp timestampInSeconds={START_TIME} />
  ));
