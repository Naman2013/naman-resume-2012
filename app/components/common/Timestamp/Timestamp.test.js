import React from 'react';
import renderer from 'react-test-renderer';
import Timestamp from './';

const START_TIME = 1499570100;

it('should render nothing, when no timestamp is provided', () => {
  const tree = renderer.create(
    <Timestamp />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render a line of text converting a timestamp in seconds to multiple timezones and utc', () => {
  const tree = renderer.create(
    <Timestamp timestampInSeconds={START_TIME} />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
