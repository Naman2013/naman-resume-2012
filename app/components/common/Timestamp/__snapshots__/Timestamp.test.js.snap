// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a line of text converting a timestamp in seconds to multiple timezones and utc 1`] = `
<span>
  11:15pm EDT
   · 
  8:15pm PDT
   · 
  03:15 UTC
</span>
`;

exports[`should render nothing, when no timestamp is provided 1`] = `null`;
